app/code/Magento/Catalog/Block/Product/ListProduct.php

```diff
@@ -466,6 +466,26 @@ private function initializeProductCollection()
 
         $this->addToolbarBlock($collection);
 
+        $restrictedIds = [];
+        $prodCollcetion = clone $collection;
+        $attributeToSelect = [];
+        foreach ($layer->getState()->getFilters() as $filterItem) {
+            $attributeToSelect[] = $filterItem->getFilter()->getRequestVar();
+        }
+        foreach ($prodCollcetion as $product) {
+            if (in_array($product->getTypeId(), ['configurable', 'group', 'bundle'])) {
```

Comment:
Additional filters for the collection should be in the corresponded modules: ConfigurableProduct, GroupedProduct, BundledProduct. Module Catalog should not know about other product types except Simple.